src/redux/sagas/onboarding.js

```diff
@@ -0,0 +1,51 @@
+import {
+  put, takeLatest, call, select,
+} from 'redux-saga/effects';
+
+import { blockchainActions, onBoardingActions } from '../actions';
+import { blockchainSelectors } from '../selectors';
+import { getComplimentaryLLD } from '../../api/backend';
+import { getBalanceByAddress } from '../../api/nodeRpcCall';
+
+// WORKERS
+
+function* claimComplimentaryLLDWorker() {
+  try {
+    const walletAddress = yield select(blockchainSelectors.userWalletAddressSelector);
+    const getComplimentaryLLDResponse = yield call(getComplimentaryLLD, walletAddress);
+    let errorData = {};
+    if (getComplimentaryLLDResponse.status === 200) {
+      // FIXME eventually we need a notification modal, not just error modal
+      errorData = { details: 'Complimentary LLDs claimed. Please refresh the page' };
+      yield put(blockchainActions.setErrorExistsAndUnacknowledgedByUser.success(true));
+      yield put(blockchainActions.setError.success(errorData));
+      yield call(getBalanceByAddress, walletAddress);
+    } else {
+      errorData = {
+        details: typeof getComplimentaryLLDResponse?.data === 'string'
+          ? getComplimentaryLLDResponse?.data
+          : 'Technical error without description, please let the devs know',
+      };
+      yield put(blockchainActions.setErrorExistsAndUnacknowledgedByUser.success(true));
+      yield put(blockchainActions.setError.success(errorData));
+    }
+  } catch {
+    const errorData = { details: 'Error in claiming, send screenshot to devs please' };
+    yield put(blockchainActions.setErrorExistsAndUnacknowledgedByUser.success(true));
+    yield put(blockchainActions.setError.success(errorData));
+  }
+}
+
+// WATCHERS
+
+function* claimComplimentaryLLDWatcher() {
+  try {
+    yield takeLatest(onBoardingActions.claimComplimentaryLLD.call, claimComplimentaryLLDWorker);
+  } catch (e) {
+    yield put(onBoardingActions.claimComplimentaryLLD.failure(e));
```

Comment:
This pretty much silently eats the error. We should:
1. log it
2. perhaps open error modal

Reply:
Okay but as far as i can see this would apply to most watchers, and would therefore be a wider task. If i m mistaken, is there an existing example i can follow

Reply:
Yup, most old watchers are broken :shrug: .

The bare-minimum would be to just add `console.error(e)` above `put`.

The way we use for new code is through blockchainWatcher. Example - requestUnregisterCompanyRegistrationWatcher added in https://github.com/liberland/liberland_frontend/pull/187 in this PR:
https://github.com/liberland/liberland_frontend/pull/168/files#diff-f29786679cd5eb47c61a53706c0199e9e9368032550145e58432493d13574699

> would therefore be a wider task

Our development is too fast and review loop too slow to do such changes to the whole codebase. We need to do it incrementally.
Older watchers have the benefit of being "battle-tested". All those that did cause issues and were eating errors got `console.error` added, for example: https://github.com/liberland/liberland_frontend/blob/d4fe5e7be49bd929770e0145d8cd6e556cadb94b/src/redux/sagas/auth.js#L60

Reply:
Solved via previous blockchainwatchers discussion